sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -331,6 +337,65 @@ private[hive] class SparkSQLCLIDriver extends CliDriver with Logging {
     console.printInfo(s"Spark master: $master, Application Id: $appId")
   }
 
+  // method body imported from Hive and translated from Java to Scala
+  override def processLine(line: String, allowInterrupting: Boolean): Int = {
```

Comment:
so the default `processLine` implementation doesn't handle `;` well? do you mean hive sql shell have this bug as well?

Reply:
yes，there is a buggy impl in hive

Reply:
I checked the hive code and seems the `;` is well handled at least in the master branch: https://github.com/apache/hive/blob/master/cli/src/java/org/apache/hadoop/hive/cli/CliDriver.java#L395

Do you mean only Hive 1.2 has this bug? Maybe we should upgrade hive.

Reply:
Yes，and I had not checked the impl on Hive master. We may judge which impl is better

Reply:
[Upgrade the built-in Hive](https://issues.apache.org/jira/browse/SPARK-23710) can fix this issue:
![image](https://user-images.githubusercontent.com/5399861/53469051-73df5680-3a97-11e9-9d9b-f3992dd3b37d.png)
Another related issue:
https://issues.apache.org/jira/browse/SPARK-12014